account/src/commonMain/kotlin/com/privateinternetaccess/account/internals/Account.kt

```diff
@@ -887,6 +900,105 @@ internal open class Account(
         }
     }
 
+    private suspend fun getDedicatedIPAsync(
+        countryCode: String,
+        regionName: String,
+        endpoints: List<AccountEndpoint>,
+        callback: (details: DedicatedIPTokenDetails?, error: List<AccountRequestError>) -> Unit
+    ) {
+        val listErrors: MutableList<AccountRequestError> = mutableListOf()
```

Comment:
This logic seems to be pretty much the same across all _Async methods.
Should we encapsulate to avoid repetition? This would be on a different ticket ofc :D

Reply:
yeah, and it's even across the K/N modules. Let's discuss it offline.